### ISSUE TITLE
Visual indication if model is being replaced

### ISSUE THREAD

**Issue by par456** [APSIMInitiative/ApsimX#9037]
### Describe the new feature

We need way to easily see if a model in our node tree is being over written by something in replacements or an experiment factor. It's so easy to edit a model thinking it'll change and not realise that it's actually being completely overwritten by a model somewhere else in the file.

I've done this plenty of times and I know other users have as well.

Any suggestions in how this should look? Maybe grey out the name? Put a symbol beside it? Hide the view in the right panel so it just displays a message that it's overwritten?

We should do some brainstorming for how this should look before going to do an implementation.

**Comment by Keith-Pembleton** [comment]
Agree that this is a needed feature (from experience).  I like the idea of putting a symbol beside it or greying out.  I don't like the idea of hiding the right panel.

**Comment by hut104** [comment]
I think it makes sense to change the RH panel to stop people changing numbers that will do nothing.
I also like the idea of grey text.
Q. what happens if it's a complex model (e.g. a whole wheat model, or a model with children)?

Beware, this sort of thing got complex in Classic UI.
________________________________
From: Andrew Paroz ***@***.***>
Sent: Wednesday, 19 June 2024 3:11 PM
To: APSIMInitiative/ApsimX ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [APSIMInitiative/ApsimX] Visual indication if model is being replaced (Issue #9037)


Describe the new feature

We need way to easily see if a model in our node tree is being over written by something in replacements or an experiment factor. It's so easy to edit a model thinking it'll change and not realise that it's actually being completely overwritten by a model somewhere else in the file.

I've done this plenty of times and I know other users have as well.

Any suggestions in how this should look? Maybe grey out the name? Put a symbol beside it? Hide the view in the right panel so it just displays a message that it's overwritten?

We should do some brainstorming for how this should look before going to do an implementation.

**Comment by par456** [comment]
That's a good point, we already use grey text under a resource to show that a model is read only, so we should use a different colour to that so it doesn't look like the same functionality.

Could we potentially hide the children of a complex model like we do for resources? Currently, if a model is replaced, all the children are also removed/replaced, so there is no merging happening there.

**Comment by hol353** [comment]
> Beware, this sort of thing got complex in Classic UI.

Yes, it is very complex and botters me a lot. 

There are a lot of ways that something can be replaced:
1. The model can be a direct child under Replacements.
2. The model can be nested under a parent model (of the same name and type) in Replacements e.g. An RUEModel under a Leaf, under a Wheat under Replacements).
3. The model can be under a CompositeFactor or nested under a model under a CompositeFactor.
4. The model can be under a Factor or nested under a model under a Factor.
5. The model can have its parameters overridden by CompositeFactor, Factor, a Manager script, a SOBOL model, a MORRIS model.
6. The model can have its parameters overridden by a Cultivar under the plant model in the simulation or cultivar under a CompositeFactor, Factor or a cultivar under Replacements or a cultivar under a plant under Replacements.

I know you're probably not talking about doing anything for cases 5 and 6 but still, this is an absolute mess. It is no wonder we still find edge cases / bugs dealing with things getting replaced e.g. #9027 merged yesterday. I would love to brainstorm a way to simplify this, e.g. can we turn the problem around so that we have one way of doing replacements, no matter if it is a simple model replacement or a factorial of replacements for experiments, morris, sobol, something else. This would have to be simpler for the user as well.

<end_rant>

**Comment by ric394** [comment]
I agree with your rant 😄. It would be nice to simplify the rules around how models are replaced.
